lib/personnummer.rb

```diff
@@ -62,4 +62,48 @@ def self.valid(str)
 
         return valid && self.testDate(year.to_i, month.to_i, day.to_i - 60)
     end
+
+    def self.format(input, longFormat = false)
+        if !self.valid(input)
+            return false
+        end
+
+        input = input.to_s
+
+        reg = /^(\d{2}){0,1}(\d{2})(\d{2})(\d{2})([\-|\+]{0,1})?(\d{3})(\d{0,1})$/;
+        match = reg.match(input)
+        century = match[1]
+        year = match[2]
+        month = match[3]
+        day = match[4]
+        sep = match[5]
+        num = match[6]
+        check = match[7]
+
+        if !(sep == '-' || sep == '+')
+          if (( century.nil? || !century.length) || Time.new.year - (century + year).to_i < 100)
+            sep = '-'
+          else
+            sep = '+'
+          end
+        end
+
+        if (century.nil? || !century.length)
+          d = Time.new
+          baseYear = 0
+
+          if sep == '+'
+            baseYear = d.year - 100
+          else
+            baseYear = d.year
+          end
+          century = "#{(baseYear - ((baseYear - year.to_i) % 100)).digits[-2..-1].reverse.join('')}"
+        end
```

Comment:
Row 71-101 should be moved to a own method so we can use the same logic in other methods, e.g `getAge` 

In the other repositories we have a function/method called `getParts` , so something like that :) https://github.com/personnummer/js/blob/master/src/personnummer.js#L48-L96

Reply:
Did consider doing that, thought I'd wait and see what you said first. On it now :)